charts/tenantfilesystem/templates/composition.yaml

```diff
@@ -0,0 +1,808 @@
+        - fromFieldPath: "spec.vpcSubnetIdC"
+          toFieldPath: "spec.forProvider.subnetID"
+          policy:
+            fromFieldPath: Required
+
+    - name: storageClass-secret
+      base:
+        apiVersion: kubernetes.crossplane.io/v1alpha1
+        kind: Object
+        metadata:
+          labels:
+            crossplane/xrd-kind: xtenantfilesystem.wadtfy.bmwgroup.net
+            crossplane/xrd-helm_chart_version: {{ .Chart.Version }}
+        spec:
+          forProvider:
+            manifest:
+              apiVersion: v1
+              kind: Secret
+              metadata:
+                name:
+                namespace:
+              type: Opaque
+              data:
+                awsRoleArn: ""
+          providerConfigRef:
+            name: provider-config-k8s
+      patches:
+        - type: CombineFromComposite
+          combine:
+            variables:
+              - fromFieldPath: "spec.product"
+              - fromFieldPath: "metadata.labels[crossplane.io/claim-namespace]"
+              - fromFieldPath: "metadata.labels[crossplane.io/claim-name]"
+            strategy: string
+            string:
+              fmt: "tenant-filesystem-%s-%s-%s-secret"
+          toFieldPath: "metadata.name"
+        - fromFieldPath: "metadata.labels[crossplane.io/claim-namespace]"
+          toFieldPath: "spec.forProvider.manifest.metadata.namespace"
+        - type: CombineFromComposite
+          combine:
+            variables:
+              - fromFieldPath: "spec.accountID"
+              - fromFieldPath: "status.roleName"
+            strategy: string
+            string:
+              fmt: "arn:aws:iam::%s:role/wadtfy/xtenantfilesystem/%s"
+          toFieldPath: "spec.forProvider.manifest.data.awsRoleArn"
+          transforms:
+            - type: string
+              string:
+                type: Convert
+                convert: ToBase64
+          policy:
+            fromFieldPath: Required
+        - type: ToCompositeFieldPath
+          fromFieldPath: "metadata.name"
+          toFieldPath: "status.storageClassSecret"
+
+    - name: storageClass
+      base:
+        apiVersion: kubernetes.crossplane.io/v1alpha1
+        kind: Object
+        metadata:
+          labels:
+            crossplane/xrd-kind: xtenantfilesystem.wadtfy.bmwgroup.net
+            crossplane/xrd-helm_chart_version: {{ .Chart.Version }}
+        spec:
+          forProvider:
+            manifest:
+              apiVersion: storage.k8s.io/v1
+              kind: StorageClass
+              metadata:
+                name:
+                annotations:
+                  wadtfy.bmwgroup.net/restricted-namespaces: ""
+              mountOptions:
+              - tls
+              - iam
+              parameters:
+                az: ""
+                csi.storage.k8s.io/provisioner-secret-name: ""
+                csi.storage.k8s.io/provisioner-secret-namespace: ""
+                directoryPerms: "700"
```

Comment:
needs to be configurable, please use old efs component as reference

Reply:
Which parameter you mean exactly ?